src/emscatter.jl

```diff
@@ -0,0 +1,296 @@
+# In this tutorial, we will learn:
+# 
+#   * How to formulate the weak form for time-harmonic electromagnetic problem
+#   * How to implement perfectly matched layers (PML)
+#   * How does periodic boundary condition work in Gridap
+# 
+# # Problem statement
+# 
+# The problem that we are trying to solve is the plane wave (TM-polarized $H_{inc}$) scattering of a cylinder dielectric (of radius $R$ and permittivity $\varepsilon$), as illustrated below. The computational cell is of height $H$ and length $L$, and we designate a perfectly matched layer (PML) thickness of $d_{pml}$.
+# 
+# ![](../assets/emscatter/Illustration.png)
+# 
+# The governing equation of this scaler electromagnetic problem in 2D is the Helmholtz equation: 
+#
+# ```math
+# \left[-\nabla\cdot\frac{1}{\varepsilon(x)}\cdot\nabla -k^2\mu(x)\right] H = f(x),
+# ```
+#
+# where $k=\omega/c$ is the wave number in free space and $f(x)$ is the source term
+# 
+# The boundary conditions are zero Dirichlet boundary on the top and bottom side $\Gamma_D$ and periodic boundary condition on the left ($\Gamma_L$) and right side ($\Gamma_R$). The PML is implemented by a coordinate streching: 
```

Comment:
I would say a few words about PML here, maybe link the wikipedia or my course notes.   But basically you need to say that what we *really* want are outgoing (radiation) boundary conditions, and PML is an artificial absorbing "layer" that emulates this in a finite domain by absorbing waves with minimal reflections (going to zero as the resolution increases).

Reply:
There are many formulations of PML.  Here, we use one of the most flexible formulations, the "stretched-coordinate" formulation, which replaces d/dx ---> 1/(1+iσ/ω)  d/dx in the PDE.   When you integrate by parts to obtain the weak form, this yields...